step/src/main/java/com/google/sps/servlets/HomepageServlet.java

```diff
@@ -35,45 +35,78 @@
 @WebServlet("/Homepage")
 public class HomepageServlet extends HttpServlet {
 
-  private static final int NUM_NOTIFS_TO_DISPLAY = 10;
-  private static final String STATUS = "status";
-  private static final String PENDING = "pending";
-  private static final String NOT_PENDING = "not pending";
+  private final int NUM_NOTIFS_TO_DISPLAY = 10;
+  private final String MATCHES = "matches";
+  private final String NOTIFICATIONS = "notifications";
+  private final String STATUS = "status";
+  private final String IMAGE = "image";
+  private final String PENDING = "pending";
+  private final String NOT_PENDING = "not pending";
```

Comment:
Can't these be static?

Reply:
These used to be static but I removed the keyword. I thought the point of static was so you could access these variables from outside the class without having to instantiate an instance of the class but here these variables aren't going to be accessed from outside of the class.

Reply:
Um well that's true but static is more than that. In Java, when a variable is static, that means it only exists in memory once, even if the class is instantiated more than once (which is part of why you can reference it without an object). When a variable isn't static, it means that every instance of the class will have an instance of that variable. After compile errors, misusing static could use up more memory than necessary but in our case it really doesn't matter :)